taskmanager/taskq.go

```diff
@@ -89,76 +96,88 @@ func (c *Client) loadTaskQ() error {
 	return nil
 }
 
-// registerTaskUsingTaskQ will register a new task using the TaskQ engine
-func (c *Client) registerTaskUsingTaskQ(task *Task) {
-
+// RegisterTask will register a new task using the TaskQ engine
+func (c *TaskManager) RegisterTask(name string, handler interface{}) (err error) {
 	defer func() {
-		if err := recover(); err != nil {
-			c.DebugLog(fmt.Sprintf("registering task panic: %v", err))
+		if panicErr := recover(); panicErr != nil {
+			err = fmt.Errorf(fmt.Sprintf("registering task panic: %v", panicErr))
```

Comment:
can't we use somehow zerolog logger here?

Reply:
I added one line after it, so it looks like this:
```golang
defer func() {
		if panicErr := recover(); panicErr != nil {
			err = fmt.Errorf(fmt.Sprintf("registering task panic: %v", panicErr))
			c.options.logger.Error().Msg(err.Error())
		}
	}()
```

I left error-returning to allow other functions to proper react to the error.